src/persist-client/src/internal/service.rs

```diff
@@ -0,0 +1,226 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+#![allow(missing_docs, dead_code)] // WIP
+#![allow(clippy::clone_on_ref_ptr, clippy::disallowed_methods)] // Generated code does this
+
+use std::collections::{BTreeMap, BTreeSet};
+use std::pin::Pin;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::{Arc, RwLock};
+use std::time::{Instant, SystemTime};
+
+use async_trait::async_trait;
+use futures::Stream;
+use mz_ore::cast::CastFrom;
+use mz_ore::now::SYSTEM_TIME;
+use prost::Message;
+use tokio::sync::mpsc::UnboundedSender;
+use tokio_stream::wrappers::UnboundedReceiverStream;
+use tokio_stream::StreamExt;
+use tonic::metadata::AsciiMetadataKey;
+use tonic::{Request, Response, Status, Streaming};
+use tracing::{debug, info, info_span, warn};
+
+use crate::internal::metrics::PubSubServerMetrics;
+use mz_persist::location::VersionedData;
+use mz_proto::{ProtoType, RustType};
+
+use crate::rpc::PERSIST_PUBSUB_CALLER_KEY;
+
+include!(concat!(
+    env!("OUT_DIR"),
+    "/mz_persist_client.internal.service.rs"
+));
+
+#[derive(Debug)]
+pub struct PersistService {
```

Comment:
I decided partway through prototyping that I think it's cleaner to make this module _only_ generated code and just push all our impls into mod rpc

Reply:
only generated code, as in only

```rust
include!(concat!(
    env!("OUT_DIR"),
    "/mz_persist_client.internal.service.rs"
));
```

and whatever traits we need to implement for those generated types?

Reply:
> whatever traits we need to implement for those generated types?

which traits?

Reply:
was meant more hypothetically, there aren't any traits we need on them right now

Reply:
gotcha. yeah, I was literally thinking the `include!` and nothing else

Reply:
oh and fun fact: rust trait coherence just means you have to impl it in the same _crate_ not necessarily the same _module_

Reply:
sure, that works for me